src/parser-includes.ts

```diff
@@ -129,14 +129,23 @@ export class ParserIncludes {
                 }
             } else if (value["component"]) {
                 const {domain, port, projectPath, componentName, ref} = this.parseIncludeComponent(value["component"]);
-                // converts component to project
-                const files = [`${componentName}.yml`, `${componentName}/template.yml`, null];
+                // converts component to project. gitlab allows two different file path ways to include a component
+                let files = [`${componentName}.yml`, `${componentName}/template.yml`, null];
+                const isLocalComponent = projectPath === `${gitData.remote.group}/${gitData.remote.project}` && ref === gitData.commit.SHA;
+
+                // If a file is present locally, keep only that one in the files array to avoid downloading the other one that never exists
+                for (const f of files) {
+                    const localFileName = `${cwd}/${stateDir}/includes/${gitData.remote.host}/${projectPath}/${ref}/${f}`;
+                    if (fs.existsSync(localFileName)) {
+                        files = [f];
+                        break;
+                    }
+                }
```

Comment:
I'm thinking about performing this only when `fetchIncludes == false` (--no-fetch-includes) , what do you think ?

Reply:
Thanks, good catch, yeah at least for the case where somehow a template is switched from `templates/mycomponent.yml`  to `mycomponent/template.yml` it would help a bit, although in that case, if the `.gitlab-ci-local/` were not cleaned you would end up with the 2 files which is as far as I saw undefined as of Gitlab documentation.  
I will do the change, supposing that fetch-includes pre-cleans each include.